src/routes/route.go

```diff
@@ -18,61 +18,57 @@ func New() *echo.Echo {
+	patient.DELETE("/:id", caHandler.Patient.RemovePatientByIDHandler, middlewares.GrantAdmin)
 
-	// polyclinic
-	polyclinic := route.Group("/polyclinics")
-	polyclinic.POST("", caHandler.Polyclinic.CreatePolyclinicHandler)
+	polyclinic := authenticatedRoute.Group("/polyclinics")
+	polyclinic.POST("", caHandler.Polyclinic.CreatePolyclinicHandler, middlewares.GrantAdmin)
 	polyclinic.GET("", caHandler.Polyclinic.ShowAllPolyclinicsHandler)
 	polyclinic.GET("/:id", caHandler.Polyclinic.ShowPolyclinicByIDHandler)
-	polyclinic.PUT("/:id", caHandler.Polyclinic.AmendPolyclinicByIDHandler)
-	polyclinic.DELETE("/:id", caHandler.Polyclinic.RemovePolyclinicByIDHandler)
+	polyclinic.PUT("/:id", caHandler.Polyclinic.AmendPolyclinicByIDHandler, middlewares.GrantAdmin)
+	polyclinic.DELETE("/:id", caHandler.Polyclinic.RemovePolyclinicByIDHandler, middlewares.GrantAdmin)
 
-	// doctor
-	doctor := route.Group("/doctors")
-	doctor.POST("", caHandler.Doctor.CreateDoctorHandler)
+	doctor := authenticatedRoute.Group("/doctors")
+	doctor.POST("", caHandler.Doctor.CreateDoctorHandler, middlewares.GrantAdmin)
 	doctor.GET("", caHandler.Doctor.ShowAllDoctorsHandler)
 	doctor.GET("/:id", caHandler.Doctor.ShowDoctorByIDHandler)
-	doctor.PUT("/:id", caHandler.Doctor.AmendDoctorByIDHandler)
-	doctor.DELETE("/:id", caHandler.Doctor.RemoveDoctorByIDHandler)
+	doctor.PUT("/:id", caHandler.Doctor.AmendDoctorByIDHandler, middlewares.GrantAdmin)
+	doctor.DELETE("/:id", caHandler.Doctor.RemoveDoctorByIDHandler, middlewares.GrantAdmin)
 
-	// nurse
-	nurse := route.Group("/nurses")
-	nurse.POST("", caHandler.Nurse.CreateNurseHandler)
+	nurse := authenticatedRoute.Group("/nurses")
+	nurse.POST("", caHandler.Nurse.CreateNurseHandler, middlewares.GrantAdmin)
 	nurse.GET("", caHandler.Nurse.ShowAllNursesHandler)
 	nurse.GET("/:id", caHandler.Nurse.ShowNurseByIDHandler)
-	nurse.PUT("/:id", caHandler.Nurse.AmendNurseByIDHandler)
-	nurse.DELETE("/:id", caHandler.Nurse.RemoveNurseByIDHandler)
+	nurse.PUT("/:id", caHandler.Nurse.AmendNurseByIDHandler, middlewares.GrantAdmin)
+	nurse.DELETE("/:id", caHandler.Nurse.RemoveNurseByIDHandler, middlewares.GrantAdmin)
 
-	// admin
-	admin := route.Group("/admins")
+	admin := authenticatedRoute.Group("/admins", middlewares.GrantAdmin)
 	admin.POST("", caHandler.Admin.CreateAdminHandler)
 	admin.GET("", caHandler.Admin.ShowAllAdminsHandler)
 	admin.GET("/:id", caHandler.Admin.ShowAdminByIDHandler)
 	admin.PUT("/:id", caHandler.Admin.AmendAdminByIDHandler)
 	admin.DELETE("/:id", caHandler.Admin.RemoveAdminByIDHandler)
 
-	// queue
-	queue := route.Group("/queues")
-	queue.POST("", caHandler.Queue.CreateQueueHandler)
+	queue := authenticatedRoute.Group("/queues")
+	queue.POST("", caHandler.Queue.CreateQueueHandler, middlewares.GrantAdmin)
 	queue.GET("", caHandler.Queue.ShowAllQueuesHandler)
 	queue.PUT("/:id", caHandler.Queue.AmendQueueByIDHandler)
 	queue.DELETE("/:id", caHandler.Queue.RemoveQueueByIDHandler)

```

Comment:
Prefer only the admin can delete the queue

Reply:
OK, https://github.com/ALTERA-CAPSTON-41/backend/pull/55/commits/f57ab04b6e6477fb40bc0650fad3c14a6ab3a1cc.